test/util/polkadot-api.ts

```diff
@@ -1,48 +1,96 @@
-import { createClient, type PolkadotClient } from "polkadot-api";
+import { createClient, type PolkadotClient, type TypedApi } from "polkadot-api";
 import { WebSocketProvider } from "polkadot-api/ws-provider/node";
 import { relaychain, storagehub } from "@polkadot-api/descriptors";
 
-export type TypesBundle = typeof relaychain | typeof storagehub;
+export type TypesBundle = typeof storagehub | typeof relaychain;
 
-export const getClient = async (endpoint: string, typesBundle: TypesBundle) => {
-  const client = createClient(WebSocketProvider(endpoint));
-  const api = client.getTypedApi(typesBundle);
-  const rt = await api.runtime.latest();
-  return { api, rt, client };
+type StorageHubApi = TypedApi<typeof storagehub>;
+
+export const waitForRandomness = async (api: StorageHubApi, timeoutMs = 60_000) => {
+  process.stdout.write("Waiting for randomness...");
+
+  const waitForValueOrTimeout = (timeoutMs: number): Promise<number> => {
+    return new Promise((resolve, reject) => {
+      const timeout = setTimeout(() => {
+        reject(new Error("Timeout"));
+      }, timeoutMs);
+
+      let valueCount = 0;
+      const subscription = api.query.Randomness.LatestOneEpochAgoRandomness.watchValue(
+        "best"
+      ).subscribe((value) => {
+        valueCount++;
+
+        if (!value) {
+          reject(new Error("Randomness value is undefined"));
+          return;
+        }
+
+        if (valueCount === 2) {
+          clearTimeout(timeout);
+          subscription.unsubscribe();
+          resolve(value[1]);
```

Comment:
What is the first item of `value` (`value[0]`)?

Reply:
`value` is `[FixedSizeBinary<32>, number] | undefined` ; which is papi's version of the rust type: Option<(H256,u32)>. So in context of the storage this would be presumably be the randomness.

Reply:
Oh, I imagine this is the representation of what in our runtime code is returned by calling `random`, which is like you said `(H256, u32)` (with the `Option` 'cause you're looking at storage directly). With that in mind, why are you returning the `blockHeight` (second value of the tuple) in `waitForRandomness`, instead of the actual randomness (first value of the tuple)?

Reply:
Fair enough, will just return both.